src/poprox_concepts/api/recommendations/v3.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from typing import TypeAlias
+
+from pydantic import BaseModel, Field, JsonValue, PositiveInt
+
+from poprox_concepts.api.recommendations.versions import ProtocolVersions, RecommenderInfo
+from poprox_concepts.domain.newsletter import Impression
+from poprox_concepts.domain.profile import InterestProfile
+from poprox_concepts.domain.recommendation import CandidateSet
+
+
+class ProtocolModelV3_0(BaseModel):
+    """
+    Version 3.0 of the POPROX protocol changed the return model of a recommendation to return a list of "sections"
+    """
+
+    protocol_version: ProtocolVersions = Field(default=ProtocolVersions.VERSION_3_0, frozen=True)
+
+
+class RecommendationRequestV3(ProtocolModelV3_0):
+    candidates: CandidateSet
+    interacted: CandidateSet
+    interest_profile: InterestProfile
+    num_recs: PositiveInt
+
+
+class RecommendationResponseSection(BaseModel):
+    title: str
+    recommendations: RecommendationList_v3
+
+
+Extra: TypeAlias = dict[str, JsonValue]
+
+
+class RecommendationList_v3(ProtocolModelV3_0):
+    articles: list[Impression] = Field(default_factory=list)
+    extras: list[Extra] = Field(default_factory=list)
+
+
+class RecommendationResponseV3(ProtocolModelV3_0):
+    recommendations: list[RecommendationResponseSection]
+    recommender: RecommenderInfo | None = Field(default=None)
```

Comment:
```suggestion
    recommender: RecommenderInfo | None = Field(default=None)
    template_name: str | None = Field(default=None)
```
This would let experiments choose their own email template -- we could make default mean "whatever the current system model is". This is also probably the much-cheaper version of *most* of the customization options I want to push elsewhere. in-so-much as I think anything you could do with my fancier options can be done with this option.

Reply:
I do want experiments to be able to choose a template, but I was envisioning that being a fixed property of an experiment determined by the manifest rather than the recommender. Since we still have the restriction that templates have to be deployed in our code in order to be used, I'm pretty hesitant to introduce a parameter that couples the state of the platform and the state of the recommender as tightly as `template_name` would here.